darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterStrategyFactoryImpl.java

```diff
@@ -82,18 +84,30 @@ public DarkClusterStrategyFactoryImpl(@Nonnull Facilities facilities,
     _random = random;
     _darkClusterDispatcher = darkClusterDispatcher;
     _verifierManager = verifierManager;
-    _rateLimiter = rateLimiter;
+    _rateLimiterFactory = rateLimiterFactory;
     _clusterListener = new DarkClusterListener();
   }
 
+  public DarkClusterStrategyFactoryImpl(@Nonnull Facilities facilities,
```

Comment:
Can we remove this constructor?

Reply:
Added a Deprecated decorator. Removing the constructor would break backward compatibility.